core/src/main/java/brooklyn/location/basic/WinRmMachineLocation.java

```diff
@@ -0,0 +1,266 @@
+
+    public static final ConfigKey<String> WINDOWS_USERNAME = ConfigKeys.newStringConfigKey("windows.username",
+            "Username to use when connecting to the remote machine");
+
+    public static final ConfigKey<String> WINDOWS_PASSWORD = ConfigKeys.newStringConfigKey("windows.password",
+            "Password to use when connecting to the remote machine");
+
+    public static final ConfigKey<Integer> COPY_FILE_CHUNK_SIZE_BYTES = ConfigKeys.newIntegerConfigKey("windows.copy.file.size.bytes",
+            "Size of file chunks (in bytes) to be used when copying a file to the remote server", 1024);
+
+    @SetFromFlag
+    protected String user;
+
+    @SetFromFlag(nullable = false)
+    protected InetAddress address;
+
+    @Override
+    public InetAddress getAddress() {
+        return address;
+    }
+
+    @Override
+    public OsDetails getOsDetails() {
+        return null;
+    }
+
+    @Override
+    public MachineDetails getMachineDetails() {
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public String getHostname() {
+        return address.getHostAddress();
+    }
+
+    @Override
+    public Set<String> getPublicAddresses() {
+        return null;
+    }
+
+    @Override
+    public Set<String> getPrivateAddresses() {
+        return null;
+    }
+
+    public WinRmToolResponse executeScript(String script) {
+        return executeScript(ImmutableList.of(script));
+    }
+
+    public WinRmToolResponse executeScript(List<String> script) {
+        Collection<Throwable> exceptions = Lists.newArrayList();
+        for (int i = 0; i < 10; i++) {
+            try {
+                WinRmTool winRmTool = WinRmTool.connect(getHostname(), getUsername(), getPassword());
+                WinRmToolResponse response = winRmTool.executeScript(script);
+                return response;
+            } catch (Exception e) {
+                LOG.warn("ignoring winrm exception and retrying:", e);
+                exceptions.add(e);
+            }
+        }
+        throw Exceptions.propagate("failed to execute shell script", exceptions);
+    }
+
+    public WinRmToolResponse executePsScript(String psScript) {
+        return executePsScript(ImmutableList.of(psScript));
+    }
+
+    public WinRmToolResponse executePsScript(List<String> psScript) {
+        Collection<Throwable> exceptions = Lists.newArrayList();
+        for (int i = 0; i < 10; i++) {
+            try {
+                return executePsScriptNoRetry(psScript);
+            } catch (Exception e) {
+                LOG.warn("ignoring winrm exception and retrying after 5 seconds:", e);
+                Time.sleep(Duration.FIVE_SECONDS);
+                exceptions.add(e);
+            }
+        }
+        throw Exceptions.propagate("failed to execute powershell script", exceptions);
+    }
+
+    public WinRmToolResponse executePsScriptNoRetry(List<String> psScript) {
```

Comment:
I'd have expected the same pattern of methods to be used for executePsScript and executeScript. Is there a reason not to?